opentelemetry/src/context.rs

```diff
@@ -381,17 +388,115 @@ impl Hasher for IdHasher {
     }
 }
 
+/// A stack for keeping track of the [`Context`] instances that have been attached
+/// to a thread.
+///
+/// The stack allows for popping of contexts by position, which is used to do out
+/// of order dropping of [`ContextGuard`] instances. Only when the top of the
+/// stack is popped, the topmost [`Context`] is actually restored.
+///
+/// The stack relies on the fact that it is thread local and that the
+/// [`ContextGuard`] instances that are constructed using it can't be shared with
```

Comment:
This talks about guards constructed using the stack, but they're just stored there and not constructed using it? Or what does the "it" here refer to?

Reply:
Yes, that is not clear. This refers to the `id` in the `guards` coming from the use of the `stack`. I'll rephrase it.